webapp/src/js/controllers/messages-content.js

```diff
@@ -30,6 +30,16 @@ angular.module('inboxControllers').controller('MessagesContentCtrl',
       message: ''
     };
 
+    const userCtx = Session.userCtx(),
+          isOnlineOnly = Session.isOnlineOnly(userCtx);
+    let updateOnChange = false;
+
+    const shouldUpdateOnChange = change => isOnlineOnly &&
+                                           updateOnChange &&
+                                           (updateOnChange === true || updateOnChange === change.id);
+
+    const setUpdateOnChange = value => updateOnChange = value;
+
```

Comment:
Not using Redux here as the state doesn't need to be communicated to other components.